### PR TITLE
add thaigov

### LINKED ISSUE #357: Create dataset loader for Thai Gov v2 Corpus
Dataloader name: `thai_gov_v2/thai_gov_v2.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?thai_gov_v2

|     Dataset| thai_gov_v2 |
|-------------|---|
| Description | The dataset consists of individual news articles, each corresponding to a unique URL at the Thai government website (https://www.thaigov.go.th/). The dataset structure is as follows: a topic header is followed by the content of the news article, which is then succeeded by a blank line and the source URL |
| Subsets | - |
| Languages | tha |
| Tasks | Language Modeling |
| License | Open Data Commons Public Domain Dedication and License (pddl) |
| Homepage | https://github.com/PyThaiNLP/thaigov-v2-corpus/tree/master/data |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by TysonYu**
Closes #357.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by SamuelCahyawijaya — APPROVED** [review]
@TysonYu : Thank you for implementing the datalaoder. 
Very smart way to download the whole repo instead of listing each file in the repo 😆.
LGTM!

**Comment by sabilmakbar** [comment]
Hi @TysonYu, a suggestion to change the init PR message of **Closes #{ISSUE_NUMBER}** so that it will be linked to the dataloader issue for coming PRs (I've done it on this one, tho).

**Review by sabilmakbar — COMMENTED** [review]
rather than having to write on `_split_generators` and re-read again in `_generate_examples`, why we don't pass the `all_data` list in `_split_generators` `gen_kwargs` and use it directly on `generate_examples`? I think passing such is possible (see [this SEACrowd Implementation](https://github.com/SEACrowd/seacrowd-datahub/blob/86a4d4cfdd55c14267fed03ed0903c6fc774df1d/seacrowd/sea_datasets/bloom_captioning/bloom_captioning.py#L231))

**Comment by TysonYu** [comment]
> Hi @TysonYu, a suggestion to change the init PR message of **Closes #{ISSUE_NUMBER}** so that it will be linked to the dataloader issue for coming PRs (I've done it on this one, tho).

Okay, will do it for later ones.

**Comment by TysonYu** [comment]
> rather than having to write on `_split_generators` and re-read again in `_generate_examples`, why we don't pass the `all_data` list in `_split_generators` `gen_kwargs` and use it directly on `generate_examples`? I think passing such is possible (see [this SEACrowd Implementation](https://github.com/SEACrowd/seacrowd-datahub/blob/86a4d4cfdd55c14267fed03ed0903c6fc774df1d/seacrowd/sea_datasets/bloom_captioning/bloom_captioning.py#L231))

Hey, I do by this way because it seems to be logically correct and clear. I agree your mentioned approach is another implementation and still my current approach should be fine. I think some other dataloaders also did in this way, such as indosum.

File: seacrowd/sea_datasets/thaigov/thaigov.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+        list_all_txt_files = list(glob.glob(os.path.join(data_dir, "thaigov-v2-corpus-master", "data", "*", "*", "*", "*.txt")))
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/thaigov/thaigov.py:127]: Done

**Review by SamuelCahyawijaya — APPROVED** [review]
@TysonYu: thank you for contributing! LGTM!

**Review by sabilmakbar — APPROVED** [review]
LGTM! Thanks, @TysonYu.

**Pull request merged**